packages/@aws-cdk/core/lib/stack-synthesizers/default-synthesizer.ts

```diff
@@ -19,7 +19,7 @@ export const BOOTSTRAP_QUALIFIER_CONTEXT = '@aws-cdk/core:bootstrapQualifier';
 /**
  * The minimum bootstrap stack version required by this app.
  */
-const MIN_BOOTSTRAP_STACK_VERSION = 4;
+const MIN_BOOTSTRAP_STACK_VERSION = 6;
```

Comment:
I was hoping we could do this bump conditionally, only if you use CDK Pipelines.

But yeah, we don't have affordances for that right now.

Reply:
I was hoping the same, but couldn't find a reasonable way to do it.

The customer experience here is certainly something we should be aiming to improve, if my understanding is correct. This means any and all customers who upgrade to CDK v.1XX.0 will need to re-bootstrap all of their environments, even if they're not using pipelines, correct?

The only alternative that (quickly) comes to mind is to switch this minimum version on a feature flag, and allow people to opt-in to the v6 bootstrap stack only if they're using pipelines. However, as soon as we need to bump it again (whether for pipelines or something else), those customers will still need to upgrade; not sure if that's worth it. :/